### PR TITLE
Update to Preview4 and include Tier3 repos

### LINKED ISSUE #1559: Re-enable tier 3 repos in 5.0 build in master
When building 5.0 in master, all repos after runtime were disabled.
Re-enable building through tier 3 repos for the .NET 5.0 build.
These include:
- aspnet-extensions
- aspnetcore-tooling
- apsnetcore
- websdk

### PR DESCRIPTION
**Pull request by dseefeld**
Update source-build to build 5.0-preview4 for current repos and add the Tier3 repos.

Fixes #1559

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -20,7 +20,7 @@
     <Platform Condition="'$(Platform)' == '' AND '$(BuildArchitecture)' == 'arm64'">$(BuildArchitecture)</Platform>
     <Platform Condition="'$(Platform)' == ''">x64</Platform>
 
-    <UseStableVersions Condition="'$(UseStableVersions)' == ''">true</UseStableVersions>
+    <UseStableVersions Condition="'$(UseStableVersions)' == ''">false</UseStableVersions>
```
Comment (dagood) [inline Directory.Build.props:23]: Should probably go away entirely, but this is already tracked at https://github.com/dotnet/source-build/issues/1627. (Noting for backlink and context on this PR.)

File: eng/SourceBuild.Tarball.targets [hunk 2]
```diff
@@ -297,20 +297,21 @@
   </Target>
 
   <Target Name="AddTarballLeakDetection"
-          Condition="'$(IncludeLeakDetection)' == 'true'">
+          Condition="'$(IncludeLeakDetection)' == 'true'"
+          DependsOnTargets="CopyTarballFiles">
```
Comment (dagood) [inline eng/SourceBuild.Tarball.targets:301]: The targets in this file should probably get order via dependenices, not the big list at the top... but I think adding this one for now is a good start.

File: eng/Versions.props [hunk 3]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <!-- Production Dependencies -->
   <PropertyGroup>
-    <PrivateSourceBuildReferencePackagesPackageVersion>1.0.0-beta.20261.1</PrivateSourceBuildReferencePackagesPackageVersion>
-    <PrivateSourceBuiltArtifactsPackageVersion>0.1.0-5.0.100-preview.1.20229.1</PrivateSourceBuiltArtifactsPackageVersion>
+    <PrivateSourceBuildReferencePackagesPackageVersion>1.0.0-beta.20329.1</PrivateSourceBuildReferencePackagesPackageVersion>
+    <PrivateSourceBuiltArtifactsPackageVersion>0.1.0-5.0.100-bootstrap.3</PrivateSourceBuiltArtifactsPackageVersion>
```
Comment (dagood) [inline eng/Versions.props:11]: Are the contents of this documented? (Specifically, what's in it and why--I think I heard it was somewhat manual?)

File: patches/arcade/0009-Add-reference-to-PlatformAbstractions.patch [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-https://github.com/dotnet/core-setup/pull/5218.
-The tarball build uses a newer version of Microsoft.Extensions.DependencyModel
-with the dependency removed, so an explicit dependency needs to be added now until
+Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk.csproj doesn't have a reference    
+to PlatformAbstractions. It builds in the production build because it has a 
+reference to Microsoft.Extensions.DependencyModel, which also brought in    
+PlatformAbstractions. That dependency has been removed with 
+https://github.com/dotnet/core-setup/pull/5218. 
+The tarball build uses a newer version of Microsoft.Extensions.DependencyModel  
+with the dependency removed, so an explicit dependency needs to be added now until
```
Comment (dagood) [inline patches/arcade/0009-Add-reference-to-PlatformAbstractions.patch:12]: Weird patch changes, solely adding trailing whitespace... rebase gone bad?

File: patches/fsharp/0004-Revert-some-F-changes-for-new-APIs.patch [hunk 5]
```diff
@@ -0,0 +1,70 @@
+From 10a406e1b68e29aa97cdcaddd354c0916b6829df Mon Sep 17 00:00:00 2001
+From: Chris Rummel <crummel@microsoft.com>
+Date: Mon, 29 Jun 2020 18:31:35 +0000
+Subject: [PATCH 4/4] Revert some F# changes for new APIs.
+
+---
```
Comment (dagood) [inline patches/fsharp/0004-Revert-some-F-changes-for-new-APIs.patch:6]: Any more info? A decent number of patches don't have much explanation.

File: patches/runtime/0012-Update-SDK-and-remove-RID-calculation-in-favor-of-Ru.patch [hunk 6]
```diff
@@ -0,0 +1,418 @@
+From 8173f8d6070e3de59864e854f9d30b156b1bbfae Mon Sep 17 00:00:00 2001
+From: Viktor Hofer <viktor.hofer@microsoft.com>
+Date: Mon, 4 May 2020 11:45:15 +0200
+Subject: [PATCH 12/15] Update SDK and remove RID calculation in favor of
+ RuntimeInformation (#35538)
```
Comment (dagood) [inline patches/runtime/0012-Update-SDK-and-remove-RID-calculation-in-favor-of-Ru.patch:5]: I'm a little confused, I assume this a cherry-pick from a future release?

File: patches/sdk/0005-Don-t-build-for-net472.patch [hunk 7]
```diff
@@ -0,0 +1,53 @@
+From 54f8bf96e5e340ae997020574f9739cb6455f121 Mon Sep 17 00:00:00 2001
+From: dseefeld <dseefeld@microsoft.com>
+Date: Sat, 20 Jun 2020 18:58:49 -0400
+Subject: [PATCH] Don't build for net472
```
Comment (dagood) [inline patches/sdk/0005-Don-t-build-for-net472.patch:4]: Don't we need to for VSCode? Just need those targeting packages?

File: repos/fsharp.proj [hunk 8]
```diff
@@ -11,6 +11,7 @@
     <BuildCommand>$(ProjectDirectory)build$(ShellExtension) $(BuildCommandArgs)</BuildCommand>
 
     <PackagesOutput>$(ProjectDirectory)artifacts/packages/$(Configuration)/Shipping/</PackagesOutput>
+    <OutputPlacementRepoApiImplemented>false</OutputPlacementRepoApiImplemented>
```
Comment (dagood) [inline repos/fsharp.proj:14]: Any history here? How are these packages getting into the source-build flow?

File: repos/runtime.common.targets [hunk 9]
```diff
@@ -22,6 +22,28 @@
                          Rid="$(TargetRid)-$(Platform)" />
   </Target>
 
+  <Target Name="CopyTools"
+          AfterTargets="Package">
```
Comment (dagood) [inline repos/runtime.common.targets:26]: Why was this moved from `runtime-portable.proj` to `runtime.common.targets`? Does this mean both builds copy it? Does the last one win, or are they disambiguated in some way? I think we do want the portable one so our previously-source-built artifacts run on as many platforms as possible.

File: Directory.Build.props (hunk 1 above)
Comment (dseefeld) [reply Directory.Build.props:23]: Agreed

File: eng/Versions.props (hunk 3 above)
Comment (dseefeld) [reply eng/Versions.props:11]: They're not documented.  It's an almagamation of previously source-built archive from 3.1 and some packages built in the production build for 5.0.  Once we get bootstrapping going, we can get a full update from 5.0.

File: patches/arcade/0009-Add-reference-to-PlatformAbstractions.patch (hunk 4 above)
Comment (dseefeld) [reply patches/arcade/0009-Add-reference-to-PlatformAbstractions.patch:12]: That is weird.  I'll fix it.

File: patches/runtime/0012-Update-SDK-and-remove-RID-calculation-in-favor-of-Ru.patch (hunk 6 above)
Comment (dseefeld) [reply patches/runtime/0012-Update-SDK-and-remove-RID-calculation-in-favor-of-Ru.patch:5]: Yes.  It's from https://github.com/dotnet/runtime/pull/35538.  I'll add that to the patch heading.

File: repos/fsharp.proj (hunk 8 above)
Comment (dseefeld) [reply repos/fsharp.proj:14]: When `OutputPlacementRepoApiImplemented` is false and `PackagesOutput` is set, the `CopyPackage` task copies the built files.  Packages weren't being copied to the output for fsharp.

File: repos/runtime.common.targets (hunk 9 above)
Comment (dseefeld) [reply repos/runtime.common.targets:26]: Yes.  Both builds copy it.  The last one wins, which is portable if it is run.  There were cases where we wanted these copied when the portable build didn't run.

File: patches/sdk/0005-Don-t-build-for-net472.patch (hunk 7 above)
Comment (dseefeld) [reply patches/sdk/0005-Don-t-build-for-net472.patch:4]: I logged https://github.com/dotnet/source-build/issues/1640 for this, since the solution is non-obvious.

File: patches/fsharp/0004-Revert-some-F-changes-for-new-APIs.patch (hunk 5 above)
Comment (dseefeld) [reply patches/fsharp/0004-Revert-some-F-changes-for-new-APIs.patch:6]: Sorry, I don't have more info on this one.  Many were copied from 3.1.  We'll need to address them during patch removal.

**Comment by dseefeld** [comment]
OSX failure issue logged here: https://github.com/dotnet/source-build/issues/1641
